### PR TITLE
added ymdk ymd09 keyboard

### PR DESCRIPTION
**Pull request by kaeltis**
## Description

I've added support for the new 9-key "YMD09" keyboard by YMDK
(https://www.aliexpress.com/i/4000510880374.html)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ymdk/ymd09/config.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+#define RGB_DI_PIN E2
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
+#define RGBLED_NUM 22
```
Comment (zvecr) [inline keyboards/ymdk/ymd09/config.h]: 22 LEDs on a 9 key board?

Comment (kaeltis) [reply keyboards/ymdk/ymd09/config.h]: My bad, used a different keyboard as an example, should be fixed now 😅

File: keyboards/ymdk/ymd09/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# YMD09
+
+![YMD09](https://ae01.alicdn.com/kf/HTB1x6BATYvpK1RjSZPiq6zmwXXai/9-Key-QMK-Hotswap-YMDK-RGB-Support-Macro-Function-Type-C-MX-Switches-Mechanical-Keyboard-Numpad.jpg)
+
+A 9-key macropad by [YMDK](https://ymdk.aliexpress.com/store/429151) with per-key RGB lighting.
+
+* Keyboard Maintainer: [Patrick Fruh](https://github.com/kaeltis)
+* Hardware Supported: YMD09
+* Hardware Availability: [AliExpress](https://www.aliexpress.com/item/4000510880374.html)
```
Comment (zvecr) [inline keyboards/ymdk/ymd09/README.md]: Link that doesnt require login
```suggestion
* Hardware Availability: [AliExpress](https://www.aliexpress.com/i/4000510880374.html)
```

File: keyboards/ymdk/ymd09/rules.mk [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```
Comment (zvecr) [inline keyboards/ymdk/ymd09/rules.mk]: To stop users having to undo the case to reboot to bootloader, lite here lets you hold down top left key while plugging in
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**